### PR TITLE
When destination SIT delivery date is changed the actual delivery dat…

### PR DESCRIPTION
**Pull request by LeDeep**
…e also changes

## Description

When a destination SIT is delivered and the date out is changed, the actual delivery date of the shipment will change as well.

## Setup

Make a `DESTINATION` SIT request. Get to the point where your destination SIT is in `DELIVERED` state with a date out. The actual delivery date should match the date out. Edit the date out and save the SIT request, and the actual delivery date will change as well.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/166049743) for this change

### CODE REVIEW

File: src/scenes/TransportationServiceProvider/ShipmentInfo.jsx [hunk 1]
```diff
@@ -194,11 +196,40 @@ class ShipmentInfo extends Component {
...
   }
 
+  componentDidUpdate(prevProps) {
+    const { storageInTransits } = this.props;
+    if (storageInTransits !== prevProps.storageInTransits) {
+      this.actualDeliveryDate();
+    }
+  }
+
+  actualDeliveryDate = () => {
```
Comment (kimallen) [inline src/scenes/TransportationServiceProvider/ShipmentInfo.jsx]: Can you name this function as an action?  Something like `resetActualDeliveryDate`

**Comment by LeDeep** [comment]
> I'm using the shipment ENTDEL to deliver the shipment and SIT. When I edit the `out_date`, I get an "unprocessable entity". Not sure if that is happening due to data issue, this PR, or a previous issue.

Just tested same shipment on my machine and worked. Maybe repopulate your database with fresh data. It still doesn't work, let me know and we can look at it.

**Comment by LeDeep** [comment]
> Can you add an integration test to check this functionality?

yes, that's a good idea. will do

File: src/scenes/TransportationServiceProvider/ShipmentInfo.jsx (hunk 1 above)
Comment (LeDeep) [reply src/scenes/TransportationServiceProvider/ShipmentInfo.jsx]: name changed

**Comment by kimallen** [comment]
I got my db back working and the actual delivery date is successfully changing when the SIT out date is edited.  However, If I go to change the actual delivery date, it shows up as the outdate.  It appears that it is actually getting edited, but it switches in the UI to whatever the SIT outdate is.

A side thing I noticed, since I know you fixed this elsewhere-- When filling out the "Release from SIT" form, the text input is bold.  If it's super quick fix on your end to fix this, it would be great if you fixed it here.  Otherwise, I can create a story for it.

**Pull request merged**